File: backend/internal/entity/quiz.go
package entity

import (
	"time"

	"gorm.io/gorm"
)

type Quiz struct {
	ID        uint           `gorm:"primaryKey" json:"id"`
	CreatedAt time.Time      `json:"createdAt"`
	UpdatedAt time.Time      `json:"updatedAt"`
	DeletedAt gorm.DeletedAt `gorm:"index" json:"-"`

	Name      string         `json:"name"`
	UserID    uint64         `json:"-" gorm:"not null;type:bigint;column:user_id"`
	Questions []QuizQuestion `json:"questions" gorm:"foreignKey:QuizID;constraint:OnDelete:CASCADE;"`
}

type QuizQuestion struct {
	ID        uint           `gorm:"primaryKey" json:"id"`
	CreatedAt time.Time      `json:"createdAt"`
	UpdatedAt time.Time      `json:"updatedAt"`
	DeletedAt gorm.DeletedAt `gorm:"index" json:"-"`

	Name     string       `json:"name"`
	Time     int          `json:"time"`
	ImageUrl string       `json:"imageUrl"`
	Choices  []QuizChoice `json:"choices" gorm:"foreignKey:QuestionID"`
	QuizID   uint
}

type QuizChoice struct {
	ID        uint           `gorm:"primaryKey" json:"id"`
	CreatedAt time.Time      `json:"createdAt"`
	UpdatedAt time.Time      `json:"updatedAt"`
	DeletedAt gorm.DeletedAt `gorm:"index" json:"-"`

	Name       string  `json:"name"`
	Correct    bool    `json:"correct" gorm:"column:is_correct"`
	ImageUrl   *string `json:"imageUrl,omitempty"`
	QuestionID uint
}

File: frontend/src/model/quiz.ts
export interface Quiz {
    id: number;
    name: string;
    questions: QuizQuestion[];
}

export interface User {
    id: number;
    email: string;
    role: 'host' | 'player';
}

export interface Player {
    id: string;
    name: string;
}

export interface QuizQuestion {
    index: any;
    id: number;
    name: string;
    time: number;
    choices: QuizChoice[];
    correctAnswerIndex: number;
    imageUrl?: string;
}

export interface QuizChoice {
    id: number;
    name: string;
    correct: boolean;
    imageUrl?: string;
}

export interface RegisterError {
    errorStatus: number;
}

export const COLORS = ["bg-pink-400", "bg-orange-200", "bg-green-200", "bg-purple-200"];